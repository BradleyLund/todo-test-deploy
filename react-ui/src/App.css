

@import url('https://fonts.googleapis.com/css2?family=Inter:wght@900&display=swap');

html,body {
  font-family: 'Inter', sans-serif !important;
  font-weight: 900 !important;
}


#loginPage {
  display:  flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 50px;
}

#noStudentsMessage {
  text-align: center;
  margin-top: 50px;
}

#aboutSection {
  text-align: center;
  margin-top: 30px;
  border: #065DD8  solid 2px;
  border-radius: 5px;
  padding: 10px;
}

.editBox {
  margin-top: 30px;
}


#lessonPageContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 50px;
}

#loadingSpinner {
   margin-top: 200px;
   text-align: center;
}

.signinButton {
  margin-top: 20px;
  text-align: center;
}

#radioButtons {
  display: flex;
  justify-content: space-evenly;
}

.navbarLink {
  color: white;
  margin-left: 40px;
  text-align: center;
  text-decoration: underline;
}


#classTable {
  padding: 20px;
  border: #065DD8  solid 2px;
  border-radius: 5px;
  box-sizing: border-box;
}

.classAbout {
  text-align: center;
  margin-top: 10px;
  font-family: 'Inter', sans-serif !important;
  font-weight: 900 !important;
  margin-bottom: 20px;

}

#difficultyList {
  margin-top: 30px;
}

#timeTaken {
  margin-bottom: 30px;
}


.button-66 {
  background-color: #0a6bff;
  border-radius: 4px;
  border: 0;
  box-shadow: rgba(1,60,136,.5) 0 -1px 3px 0 inset,rgba(0,44,97,.1) 0 3px 6px 0;
  box-sizing: border-box;
  color: #fff;
  cursor: pointer;
  display: inherit;
  /* font-family: "Space Grotesk",-apple-system,system-ui,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol"; */
  font-size: 18px;
  font-weight: 700;
  line-height: 24px;
  margin: 0;
  width: 100%;
  padding: 16px 20px;
  position: relative;
  text-align: center;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  vertical-align: baseline;
  transition: all .2s cubic-bezier(.22, .61, .36, 1);
  margin-top: 20px;
}

.button-66:hover {
  background-color: #065dd8;
  transform: translateY(-2px);
}


