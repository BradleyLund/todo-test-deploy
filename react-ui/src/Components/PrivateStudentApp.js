import "../App.css";
import React from "react";
import axios from "axios";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import LessonPage from "./LessonPage";
import LessonHistoryTable from "./LessonHistoryTable";

class PrivateStudentApp extends React.Component {
  constructor(props) {
    super(props);

    // initialise the state:
    this.state = {
      isLoading: false,
      lessonHistoryArray: [],
      difficultyLevel: null,
    };
  }

  componentDidMount() {
    // send a get request to the backend with the token
    this.setState({ isLoading: true });
    let access_token = window.localStorage.getItem("AuthToken");
    axios
      .get("/getstudenthistory", {
        headers: {
          Authorization: `Bearer ${access_token}`,
        },
      })
      .then((res) => {
        // getting the list of lessons once the student has logged in and been authenticated
        this.setState({
          lessonHistoryArray: res.data.lessonHistoryArray,
          isLoading: false,
          difficultyLevel: res.data.difficultyLevel,
        });
      })
      .catch((error) => {
        console.error(error);
        this.setState({ isLoading: false });
      });
  }

  render() {
    return (
      <div>
        <Router>
          <Navbar bg="primary" expand="lg" variant="dark">
            <Container>
              <Navbar.Brand>Welcome, {this.props.username} </Navbar.Brand>
              <Navbar.Toggle aria-controls="basic-navbar-nav" />
              <Navbar.Collapse id="basic-navbar-nav">
                <Nav>
                  <Navbar.Text>
                    <Link to="/">
                      <span className="navbarLink">Home</span>
                    </Link>
                  </Navbar.Text>
                  <Navbar.Text>
                    <Link to="/nextlesson">
                      <span className="navbarLink">Start your next lesson</span>
                    </Link>
                  </Navbar.Text>
                  <Nav.Link onClick={this.props.handleLogout}>
                    <span className="navbarLink">Log out</span>
                  </Nav.Link>
                </Nav>
              </Navbar.Collapse>
            </Container>
          </Navbar>

          <Switch>
            <Route path="/nextlesson">
              <LessonPage
                username={this.props.username}
                difficultyLevel={this.state.difficultyLevel}
              />
            </Route>

            {/* home needs to be at the bottom as the switch displays the first route that matches the link */}
            <Route path="/">
              <div id="classTable">
                <h2 className="classAbout">
                  Welcome to your lesson History! Here you can see all of your
                  results from previous lessons! üë®‚Äçüéì
                </h2>
                <LessonHistoryTable
                  lessonHistoryArray={this.state.lessonHistoryArray}
                />
              </div>
            </Route>
          </Switch>
        </Router>
      </div>
    );
  }
}

export default PrivateStudentApp;
