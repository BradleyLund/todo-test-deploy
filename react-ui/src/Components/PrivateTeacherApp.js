import "../App.css";
import React from "react";
import axios from "axios";
import Navbar from "react-bootstrap/Navbar";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import AddStudent from "./AddStudent";
import EditStudent from "./EditStudent";
import DifficultyExamples from "./DifficultyExamples";
import StudentTable from "./StudentTable";
import Spinner from "react-bootstrap/Spinner";

class PrivateTeacherApp extends React.Component {
  constructor(props) {
    super(props);

    // initialise the state:
    this.state = {
      studentsArray: [],
      isLoading: false,
    };
  }

  componentDidMount() {
    // send a get request to the backend with the token
    this.setState({ isLoading: true });
    let access_token = window.localStorage.getItem("AuthToken");
    axios
      .get("/getteachersclass", {
        headers: {
          Authorization: `Bearer ${access_token}`,
        },
      })
      .then((res) => {
        // getting the list of students once the teacher has logged in
        this.setState({ studentsArray: res.data });
        this.setState({ isLoading: false });
      })
      .catch((error) => {
        console.error(error);
        this.setState({ isLoading: false });
      });
  }

  render() {
    return (
      <div>
        <Router>
          <Navbar bg="primary" expand="lg" variant="dark">
            <Container>
              <Navbar.Brand>Welcome, {this.props.username} </Navbar.Brand>
              <Navbar.Toggle aria-controls="basic-navbar-nav" />

              <Navbar.Collapse id="basic-navbar-nav">
                <Nav>
                  <Navbar.Text>
                    <Link to="/">
                      <span className="navbarLink">Home</span>
                    </Link>
                  </Navbar.Text>
                  <Navbar.Text>
                    <Link to="/addstudent">
                      <span className="navbarLink">Add a student</span>
                    </Link>
                  </Navbar.Text>

                  <Navbar.Text>
                    <Link to="/editstudent">
                      <span className="navbarLink">Edit a student</span>
                    </Link>
                  </Navbar.Text>

                  <Navbar.Text>
                    <Link to="/difficultyexamples">
                      <span className="navbarLink">Difficulty Examples</span>
                    </Link>
                  </Navbar.Text>

                  <Nav.Link onClick={this.props.handleLogout}>
                    <span className="navbarLink">Log out</span>
                  </Nav.Link>
                </Nav>
              </Navbar.Collapse>
            </Container>
          </Navbar>

          <Switch>
            <Route path="/addstudent">
              <AddStudent teacherID={this.props.teacherID} />
            </Route>
            <Route path="/editstudent">
              <EditStudent studentsArray={this.state.studentsArray} />
            </Route>
            <Route path="/difficultyexamples">
              <DifficultyExamples />
            </Route>
            {/* home needs to be at the bottom as the switch displays the first route that matches the link */}
            <Route path="/">
              {this.state.isLoading ? (
                <div id="loadingSpinner">
                  <Spinner animation="border" role="status" variant="primary">
                    <span className="visually-hidden">Loading...</span>
                  </Spinner>
                </div>
              ) : this.state.studentsArray.length > 0 ? (
                <div id="classTable">
                  <h2 className="classAbout">
                    Welcome to your class! Here you can monitor all of your
                    students. üë®‚Äçüéì
                  </h2>
                  <StudentTable studentsArray={this.state.studentsArray} />
                </div>
              ) : (
                <div id="noStudentsMessage">
                  You don't have any students in your class yet, click add a
                  student
                </div>
              )}
            </Route>
          </Switch>
        </Router>
      </div>
    );
  }
}

export default PrivateTeacherApp;
