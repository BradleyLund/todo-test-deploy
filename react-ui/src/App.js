import "./App.css";
import PrivateTeacherApp from "./Components/PrivateTeacherApp";
import Login from "./Components/Login";
import React from "react";
import axios from "axios";
import "bootstrap/dist/css/bootstrap.min.css";
import Spinner from "react-bootstrap/Spinner";
import PrivateStudentApp from "./Components/PrivateStudentApp";

class App extends React.Component {
  constructor(props) {
    super(props);

    // initialise the state:
    this.state = {
      loggedin: null,
      username: "",
      isTeacher: null,
      teacherID: "",
      isLoading: false,
    };
  }

  componentDidMount() {
    if (window.localStorage.getItem("AuthToken") !== null) {
      // check the authtoken on the backend, and if the auth token is valid set loggedin to true otherwise say not logged in
      // and if authorized make sure what type of authorization, teacher or student. then display different private apps depending
      this.setState({ isLoading: true });

      let access_token = window.localStorage.getItem("AuthToken");
      axios
        .get("/authorize", {
          headers: {
            Authorization: `Bearer ${access_token}`,
          },
        })
        .then((res) => {
          // handling getting the initial data once the user is logged in
          this.setState({
            username: res.data.username,
            isTeacher: res.data.isTeacher,
            loggedin: true,
            teacherID: res.data.teacherID,
            studentID: res.data.studentID,
          });
          this.setState({ isLoading: false });
        })
        .catch((error) => {
          console.error(error);
          this.setState({ isLoading: false });
        });
    } else {
      this.setState({ username: "", isTeacher: null, loggedin: false });
    }
  }

  handleLogout = (event) => {
    event.preventDefault();

    // remove the AuthToken from localStorage when logging out
    window.localStorage.removeItem("AuthToken");
    window.location.reload();
  };
  // here we want to put if logged in then show private app otherwise show the login screen
  // we need to authorize the token properly.
  render() {
    return (
      <div id="parentDiv">
        {this.state.isLoading ? (
          <div id="loadingSpinner">
            <Spinner animation="border" role="status" variant="primary">
              <span className="visually-hidden">Loading...</span>
            </Spinner>
          </div>
        ) : this.state.loggedin ? (
          this.state.isTeacher ? (
            <PrivateTeacherApp
              handleLogout={this.handleLogout}
              username={this.state.username}
              teacherID={this.state.teacherID}
            />
          ) : (
            <PrivateStudentApp
              handleLogout={this.handleLogout}
              username={this.state.username}
            />
          )
        ) : (
          <Login />
        )}
      </div>
    );
  }
}

export default App;
