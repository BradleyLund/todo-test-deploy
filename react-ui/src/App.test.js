import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from "react-test-renderer";
import PrivateStudentApp from "./Components/PrivateStudentApp";
import PrivateTeacherApp from "./Components/PrivateTeacherApp";

// A frontend test to see whether the login button is rendered

test("renders login button", () => {
  render(<App />);
  const linkElement = screen.getByText(/Log In/i);
  expect(linkElement).toBeInTheDocument();
});

//  test the render tree of the private student app

it("renders Private Student App correctly", () => {
  const tree = renderer.create(<PrivateStudentApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
