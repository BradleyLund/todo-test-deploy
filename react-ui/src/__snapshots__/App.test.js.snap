// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Private Student App correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      className="container"
    >
      <span
        className="navbar-brand"
      >
        Welcome, 
         
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="navbar-text"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Home
              </span>
            </a>
          </span>
          <span
            className="navbar-text"
          >
            <a
              href="/nextlesson"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Start your next lesson
              </span>
            </a>
          </span>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="navbarLink"
            >
              Log out
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    id="classTable"
  >
    <h2
      className="classAbout"
    >
      Welcome to your lesson History! Here you can see all of your results from previous lessons! üë®‚Äçüéì
    </h2>
    <div
      className="table-responsive"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Date
            </th>
            <th>
              Difficulty Level
            </th>
            <th>
              Score
            </th>
            <th>
              Percentage
            </th>
            <th>
              Total Time Spent
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;

exports[`renders Private Teacher App correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <div
      className="container"
    >
      <span
        className="navbar-brand"
      >
        Welcome, 
         
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <span
            className="navbar-text"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Home
              </span>
            </a>
          </span>
          <span
            className="navbar-text"
          >
            <a
              href="/addstudent"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Add a student
              </span>
            </a>
          </span>
          <span
            className="navbar-text"
          >
            <a
              href="/editstudent"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Edit a student
              </span>
            </a>
          </span>
          <span
            className="navbar-text"
          >
            <a
              href="/difficultyexamples"
              onClick={[Function]}
            >
              <span
                className="navbarLink"
              >
                Difficulty Examples
              </span>
            </a>
          </span>
          <a
            className="nav-link"
            data-rr-ui-event-key={null}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="navbarLink"
            >
              Log out
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    id="loadingSpinner"
  >
    <div
      className="spinner-border text-primary"
      role="status"
    >
      <span
        className="visually-hidden"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
